dotCMS/src/main/java/com/dotmarketing/portlets/workflows/util/WorkflowActionletUtil.java

```diff
@@ -0,0 +1,181 @@
+package com.dotmarketing.portlets.workflows.util;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClassParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowHistory;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+import com.liferay.util.Validator;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.StringTokenizer;
+
+/**
+ * This utility class provides common-use methods that can be accessed by the different workflow
+ * actionlets provided either OOTB by dotCMS, or added via Static or Dynamic Plugins.
+ *
+ * @author Jose Castro
+ * @version 4.3.0
+ * @since Jan 9, 2018
+ */
+public class WorkflowActionletUtil {
+
+    private WorkflowActionletUtil() {
+
+    }
+
+    /**
+     * Retrieves the list of {@link User} objects from a list of IDs represented as a String. This
+     * list can be composed of user IDs, email addresses, or role keys. For role keys, all the users
+     * assigned to those roles will be included in the result user list. The tokens will be split by
+     * the specified delimiter value.
+     *
+     * @param ids       The user IDs, email addresses, or role keys that will be used to retrieve
+     *                  the
+     *                  {@link User} objects.
+     * @param delimiter The delimiter character used to separate each token in the String.
+     *
+     * @return The list of {@link User} objects based on the specified IDs.
+     */
+    public static Set<User> getUsersFromIds(final String ids, final String delimiter) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)